profitbricks/src/test/java/org/jclouds/profitbricks/features/DataCenterApiMockTest.java

```diff
@@ -0,0 +1,236 @@
+      try {
+         DataCenter dataCenter = api.getDataCenter(id);
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(dataCenter);
+         assertEquals(dataCenter.id(), id);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testGetNonExistingContainer() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setResponseCode(500).setBody(payloadFromResource("/fault-404.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      DataCenterApi api = pbApi.dataCenterApi();
+
+      String id = "random-non-existing-id";
+      try {
+         DataCenter dataCenter = api.getDataCenter(id);
+      } catch (ResourceNotFoundException ex) {
+         assertRequestHasCommonProperties(server.takeRequest());
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testGetDataCenterState() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/datacenter/datacenter-state.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      DataCenterApi api = pbApi.dataCenterApi();
+
+      String id = "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee";
+      try {
+         ProvisioningState state = api.getDataCenterState(id);
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(state);
+         assertEquals(state, ProvisioningState.AVAILABLE);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testCreateDataCenter() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/datacenter/datacenter-created.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      DataCenterApi api = pbApi.dataCenterApi();
+
+      try {
+         DataCenter dataCenter = api.createDataCenter(
+                 DataCenter.Request.CreatePayload.create("JClouds-DC", Location.DE_FRA)
+         );
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(dataCenter);
+         assertEquals(dataCenter.id(), "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee");
+         assertEquals(dataCenter.version(), 1);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testCreateDataCenterWithIllegalArguments() throws Exception {
+      String[] names = {"JCl@ouds", "JC|ouds", "^clouds", ""};
+      for (String name : names)
+         try {
+            DataCenter.Request.CreatePayload.create(name, Location.US_LAS);
+            fail("Should have failed for name: ".concat(name));
+         } catch (Exception ex) {
+            assertTrue(ex instanceof IllegalArgumentException);
```

Comment:
Hmm. I was thinking that if (for some reason) <code>CreatePayload#create</code> throws a different kind of exception, it would skip the <code>fail()</code> and <code>catch</code> clause; thus, the test becoming a success. So... should just <code>IllegalArgumentException</code> be okay?